fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/PredicatedTransformer.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.biome;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+import it.unimi.dsi.fastutil.doubles.DoubleList;
+
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.layer.util.LayerRandomnessSource;
+
+import net.fabricmc.fabric.api.biomes.v1.BiomeNeighborsPredicate;
+
+final class PredicatedTransformer {
+	private final List<PredicatedBiomeEntry> predicates = new ArrayList<>();
+
+	void addPredicatedBiome(Biome biome, BiomeNeighborsPredicate predicate, double chance) {
+		predicates.add(new PredicatedBiomeEntry(biome, predicate, chance));
+	}
+
+	Biome transform(Biome biome, LayerRandomnessSource random, Collection<Biome> borders) {
+		if (predicates.isEmpty()) {
+			return biome;
+		} else if (predicates.size() == 1) {
```

Comment:
No need for `else` after a `return` statement

Reply:
IMO this makes it more explicit/clear. It allows one to quickly see the exclusive nature of the branches immediately before seing the returns.

Reply:
Nah, `else` brings extraneous indent, which outweighs the benefits you stated given we use 4-space-sized tab indents. The code body shifts fast and confuses people.

Reply:
It would have to be indented anyways since it's an if statement???? I don't understand where you're coming from with your comment.

Reply:
I mean you can just emit the `else` here and the code is still fine

Reply:
But it still makes the code more explicit. I don't think that the `} else if (` as opposed to an `if (` makes the code *less* readable like you are saying.

Reply:
more people have expressed a preference for `else`